shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -77,10 +77,12 @@
 @NotThreadSafe
 @Slf4j
 public class CommandEncoder extends FlushingMessageToByteEncoder<Object> {
+    private static final int INFINITE_SIZE = Integer.MAX_VALUE;
+    private static final Duration EVICTION_TIME = Duration.ofDays(1);
     private static final byte[] LENGTH_PLACEHOLDER = new byte[4];
     private final Function<Long, AppendBatchSizeTracker> appendTracker;
     private final MetricNotifier metricNotifier;
-    private final Map<Map.Entry<String, UUID>, Session> setupSegments = new HashMap<>();
+    private final SimpleCache<String, Session> setupSegments = new SimpleCache<>(INFINITE_SIZE, EVICTION_TIME, null);
```

Comment:
These values can't be dropped randomly. So a cache is not an appropriate data structure. If one of these goes missing a write will suddenly have all their appends fail.

Reply:
got it!